app/uk/gov/hmrc/teamsandrepositories/connectors/JenkinsConnector.scala

```diff
@@ -92,11 +92,11 @@ class JenkinsConnector @Inject()(
     httpClientV2
       .get(url)
       .setHeader("Authorization" -> authorizationHeader)
-      .execute[JenkinsRoot]
+      .execute[Option[JenkinsRoot]]
       .recoverWith {
         case NonFatal(ex) =>
           logger.error(s"An error occurred when connecting to $url: ${ex.getMessage}", ex)
-          Future.failed(ex)
+          Future.successful(None)
```

Comment:
I think the client should decide if it wants to handling missing jenkins job or failure to get the job the same - some contexts it can be misleading to just say the job doesn't exist.

Reply:
A client doesn't call this it is called by the scheduled job to reload the jenkins cached data.

Reply:
NB I'm just double checking that the scheduled job reload does not delete data, but only upsert.

Reply:
Can confirm that nothing is deleted and only new/changed data is inserted. The current situation is that if one call fails then no records are updated/created until the next scheduled rebuild, this change will make it that only the failed call(s) will not be updated. The other day we had 4 hours where nothing was updated as at least one call timed out on each run.

Reply:
By `client`, I mean the `caller` of the function - normally the caller would call `.recover` as appropriate (the context you provided is all caller context). The signature `Future[Option]` is usually understood to mean optional result - failures in the Future.
If we're disguising this, I think we should at least change the `error` level of the log - since it's now BAU.